component/mcsd/updater.go

```diff
@@ -78,12 +79,16 @@ func buildUpdateTransaction(tx *fhir.Bundle, entry fhir.BundleEntry, allowedReso
 		return resourceType, nil
 	}
 
-	updateResourceMeta(resource, *entry.FullUrl)
+	originalResourceID, ok := resource["id"].(string)
```

Comment:
We can use the fhir util with `ExtractResourceInfo` for the id and resourceType etc, might need to extend the logic for the `updateResourceMeta()` though